sycl/include/CL/sycl/detail/kernel_program_cache.hpp

```diff
@@ -0,0 +1,95 @@
+//==--- kernel_program_cache.hpp - Cache for kernel and program -*- C++-*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/locked.hpp>
+#include <CL/sycl/detail/os_util.hpp>
+#include <CL/sycl/detail/pi.hpp>
+
+#include <atomic>
+#include <condition_variable>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <type_traits>
+
+namespace cl {
+namespace sycl {
+namespace detail {
+class kernel_program_cache {
```

Comment:
Please follow llvm codying guideline kernel_program_cache -> KernelProgramCache

Reply:
Other classes in this directory are named with snake_case, e.g. alligned_allocator, buffer_impl, etc.

Reply:
Probably we should rename them.